lib/common/client/middleware/request/multipart_request.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+require 'rack/mime.rb'
+
+module Common
+  module Client
+    module Middleware
+      module Request
+        class MultipartRequest < Faraday::Middleware
+          include Rack::Mime
+
+          def initialize(app)
+            super(app)
+          end
+
+          def call(env)
+            return @app.call(env) unless env[:body].is_a?(Hash)
+
+            @io_keys = []
```

Comment:
Since @io_keys is the concern of this method, I think it's an antipattern to have the other pattern be aware of the instance variable and fill it in. I'd not use an instance variable, but pass a list as an argument to the io_object_for method and have it append to the list as needed.